fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateMaterializedViewCommand.java

```diff
@@ -0,0 +1,157 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.plans.commands;
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.StmtType;
+import org.apache.doris.analysis.TableName;
+import org.apache.doris.catalog.Column;
+import org.apache.doris.catalog.DatabaseIf;
+import org.apache.doris.catalog.Env;
+import org.apache.doris.catalog.MaterializedIndexMeta;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.ScalarType;
+import org.apache.doris.catalog.TableIf;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.util.Util;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.commands.info.TableNameInfo;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ShowResultSet;
+import org.apache.doris.qe.ShowResultSetMetaData;
+import org.apache.doris.qe.StmtExecutor;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * Represents the command for SHOW CREATE MATERIALIZED VIEW.
+ */
+public class ShowCreateMaterializedViewCommand extends Command implements NoForward {
+    private static final ShowResultSetMetaData MATERIALIZED_VIEW_META_DATA =
+            ShowResultSetMetaData.builder()
+                    .addColumn(new Column("TableName", ScalarType.createVarchar(20)))
+                    .addColumn(new Column("ViewName", ScalarType.createVarchar(30)))
+                    .addColumn(new Column("CreateStmt", ScalarType.createVarchar(500)))
+                    .build();
+
+    private final String dbName;
+    private final String mvName;
+    private final String tableName;
+    private final String catalog;
+
+    public ShowCreateMaterializedViewCommand(String mvName, TableNameInfo tableNameInfo) {
+        super(PlanType.SHOW_CREATE_MATERIALIZED_VIEW_COMMAND);
+        this.tableName = tableNameInfo.getTbl();
+        this.dbName = tableNameInfo.getDb();
+        this.catalog = tableNameInfo.getCtl();
+        this.mvName = mvName;
+    }
+
+    private void validate(ConnectContext ctx) throws AnalysisException {
+        TableName tbl = new TableName(catalog, dbName, tableName);
+
+        Analyzer analyzer = new Analyzer(ctx.getEnv(), ctx);
+        tbl.analyze(analyzer);
```

Comment:
should not create Analyzer here. you can use TableNameInfo as member variable, and call tableNameInfo.validate(ctx) to do the validation

Reply:
done.